lib/membrane_opus/parser.ex

```diff
@@ -1,79 +1,375 @@
 defmodule Membrane.Opus.Parser do
   @moduledoc """
-  Parses self-delimiting Opus stream.
+  Parses a raw incoming Opus stream and adds caps information, as well as metadata.
 
-  The self-delimiting version of Opus packet contains all the information about frame sizes
-  needed to separate the packet from byte stream and decode it. This parser drops the last frame size
-  from each packet header, converting packets to the basic version. The remaining frame size needs
-  to be derived from entire packet size.
-  See https://tools.ietf.org/html/rfc6716#appendix-B for details.
+  Adds the following metadata:
+
+  duration :: non_neg_integer()
+    Number of nanoseconds encoded in this packet
   """
+
   use Membrane.Filter
 
+  import Membrane.Time
+
   alias Membrane.{Buffer, Opus, RemoteStream}
-  alias Membrane.Opus.PacketUtils
+
+  def_options delimitation: [
+                spec: :delimit | :undelimit | :keep,
+                default: :keep,
+                description: """
+                If input is delimitted (as indicated by the `self_delimiting?`
+                field in %Opus) and `:undelimit` is selected, will remove delimitting.
+
+                If input is not delimitted and `:delimit` is selected, will add delimitting.
+
+                If `:keep` is selected, will not change delimiting.
+
+                Otherwise will raise.
+
+                See https://tools.ietf.org/html/rfc6716#appendix-B for details
+                on the self-delimitting Opus format.
+                """
+              ],
+              input_delimitted?: [
```

Comment:
i couldn't see a way to get around this, unless you want users to use two `Parsers` in sequence. what do you think? this is pretty gross as a user interface, IMO